virt/virtual_machines/virt-accessing-vm-ssh.adoc

```diff
@@ -0,0 +1,84 @@
+:_content-type: ASSEMBLY
+[id="virt-accessing-vm-ssh"]
+= Using SSH to access virtual machines
+include::_attributes/common-attributes.adoc[]
+:context: virt-accessing-vm-ssh
+
+toc::[]
+
+You can configure an SSH connection to access virtual machines (VMs) by using the following methods:
+
+* `virtctl ssh` command
++
+You add a public SSH key to a VM before it boots for the first time. Then, you connect to the VM by using the `virtctl ssh` command.
```

Comment:
The key could have been added in other ways too. Doesn't have to be before the boot.

Reply:
Can you tell me how? I did not succeed in getting SSH to work, either with a service or with virtctl, unless I added the key before initial boot.

Reply:
I see that it is possible on the InstanceTypes tab. That is documented in the 4.14 docs.

Reply:
For example the VM image could have been customized or someone connected to it via password and configured an SSH key themselves. Essentially there are endless ways to configure an SSH key, using the access credentials API is just one way of accomplishing that. `virtctl ssh` is agnostic to the way the SSH key was configured in the target VM, I would just drop mentioning that it has to be configured before the first boot. It just needs to be configured.

Reply:
While I acknowledge what @0xFelix says on a technical level, I vote for explicitly document only the flows based on the automatically installed rhel9 images. Starting with long yaml now, and introducing virtctl based yaml via https://issues.redhat.com/browse/CNV-30930 in 4.15 . These flows would be easy to cover by tests.